src/main/java/com/splunk/logging/HttpEventCollectorSender.java

```diff
@@ -99,13 +100,17 @@ public HttpEventCollectorSender(
             long delay, long maxEventsBatchCount, long maxEventsBatchSize,
             String sendModeStr,
             Map<String, String> metadata) {
-        this.url = Url + HttpEventCollectorUriPath;
         this.token = token;
         this.channel = channel;
         this.type = type;
 
-        if ("Raw".equalsIgnoreCase(type)) {
-            this.url = Url + HttpRawCollectorUriPath;
+        if ("Raw".equalsIgnoreCase(type) && channel != null && !channel.trim().equals("")) {
```

Comment:
This looks good, but if the channel is not specified, it gets all the way to Splunk and fails. Mind putting in a check before this that throws and InvalidArgumentException if type is raw but the channel is not set?

Reply:
@zenmoto Updated!